src/rgw/rgw_tools.cc

```diff
@@ -47,6 +47,11 @@ int rgw_get_system_obj(RGWRados *rgwstore, RGWObjectCtx& obj_ctx, rgw_bucket& bu
   int request_len = READ_CHUNK_LEN;
   rgw_obj obj(bucket, key);
 
+  obj_version original_readv;
```

Comment:
1. Let's initialize original_readv.
2. if (!original_readv.empty()))	//String is empty=> .empty() return 0
   {
    string empty; return;
   }

   Just thinking, Why can't we return from 
   if (objv_tracker && !objv_tracker->read_version.empty()) {
     original_readv = objv_tracker->read_version;
     <<<<return (value)>>>>
   }
   This will save processing code inside do-while before rop.read().